aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -753,6 +754,31 @@ func resourceAviatrixTransitGateway() *schema.Resource {
 					},
 				},
 			},
+			"interface_mapping": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of interface names mapped to interface types and indices.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Interface name (e.g., 'eth0', 'eth1').",
```

Comment:
How flexible is name?  Must it be eth\d+$?  If so I would add a validation function 

ValidateFunc: validation.StringMatch(regexp.MustCompile(`^eth\d+$`)

Or if we think this needs to be flexible leave as is.

Reply:
for other devices this name might look different so keeping it flexible for now.